src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBinaryWriter.cs

```diff
@@ -624,19 +604,19 @@ public unsafe override void WriteText(string value)
             }
         }
 
-        public unsafe override void WriteText(char[] chars, int offset, int count)
+        private unsafe void WriteText(Span<char> chars)
```

Comment:
I did try using ReadOnlySpan but then it looked like to many methods called this overload instead of the string version so I used span instead to avoid extra string allocations.

However it might be possibly to merge this implementation with the string based method if using ReadOnlyMemory instead since it allows trying to retrieve the original string.

Reply:
> many methods called this overload instead of the string version

That does not sound right. If you have a string and there is a string overload, the compiler should pick the string overload.

Reply:
changed the code, will have to decompile the code and have a second look. Visual studio (17.2) still reports many references to the ReadOnlySpan version

Reply:
The unexpected calls are still there in the compiled assembly, should I keep it?
You might know who to involve to find out why it is called instead of the string overload.

![image](https://user-images.githubusercontent.com/3167509/178047885-5a26d349-296f-43cf-bf7a-6e184927e0ef.png)

Reply:
Hmm, this surprises me:
[SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAFxFANgHwAICYCMBYAKEIwGYACAZwFMIUqATM7MgYTJDICEJrCBvQmSFNyGACxkAogA8IYAA50AFBhwAGSgEoyAXgB8ZALJKKmgNyDhloaTIB7AG5UATs6j0qTCcdUbTZPjIAX2sRLyMlACUaegB5GBQATwBleQgYAB4AYwALCGcDf0CQghLiUSwuHip+UNsHKGcEAFdacJ91LQDgwiCgA=)
For some reason the WriteText(string) overload being an override is causing it to lose out to the span-based overload.

Reply:
I solved this by creating a separate "implementation metod" which is not an override  (I called it `WriteTextImpl` to follow what seems to be the current naming convention for implementation methods)